### PR TITLE
docker: add some tweaks  to nvidia docker

### PR DESCRIPTION
**Pull request by mloubout**
From @kenhester suggestions

Should fix issues with #2023 as cupy-cuda12x seems to install and work fine inside the image

To be discussed, do we keep nvidia-clang in CI

### CODE REVIEW

**Comment by FabioLuporini** [comment]
> To be discussed, do we keep nvidia-clang in CI

Probably we can drop it

NVC for openmp offloading on NVidia cards
Amdclang for openmp offloading on AMD cards
icx for openmp offloading on Intel cards

That's completely fine by me

IIRC, we're still stuck with our own fork of clang, so one more reason to drop it. 

Can we try again NVC + openmp offloading, now that we finally have NVC 23.5 ?

**Review by georgebisbas — COMMENTED** [review]
I would vote for dropping as well...not popular, not faster, cumbersome for non-docker....
But will we also drop related infrastructure?

**Comment by mloubout** [comment]
>Can we try again NVC + openmp offloading

Yes


Ok removing clang build and replacing by nvc

**Review by FabioLuporini — APPROVED** [review]

File: .github/workflows/docker-devito.yml [hunk 1]
```diff
@@ -24,9 +24,9 @@ jobs:
             test: 'tests/test_gpu_openacc.py tests/test_gpu_common.py'
             runner: ["self-hosted", "nvidiagpu"]
 
-          - base: 'bases:nvidia-clang'
-            tag: 'nvidia-clang'
-            flag: '--gpus all'
+          - base: 'bases:nvidia-nvc'
+            tag: 'nvidia-nvc-omp'
+            flag: '--gpus all --env DEVITO_LABGUAGE=openmp'
```
Comment (georgebisbas) [inline .github/workflows/docker-devito.yml:29]: LAN*

**Pull request merged**